app/src/main/java/com/dragic/gamehunter/viewmodel/ThemeViewModel.kt

```diff
@@ -0,0 +1,48 @@
+package com.dragic.gamehunter.viewmodel
+
+import android.util.Log
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import com.dragic.gamehunter.settings.UserSettings
+import com.dragic.gamehunter.view.theme.AppTheme
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class ThemeViewModel @Inject constructor(
+    private val userSettings: UserSettings
+) : ViewModel() {
+
+    val appTheme = mutableStateOf(userSettings.theme)
+
+    var useDarkColor by mutableStateOf(
+        when (appTheme.value) {
+            AppTheme.LIGHT -> false
+            AppTheme.DARK -> true
+        }
+    )
```

Comment:
Do you need both of these? It looks like you could achieve the same thing with just

```kt
var appTheme by mutableStateOf(userSettings.theme)
[...]
GameHunterTheme(darkTheme = themeViewModel.appTheme == AppTheme.DARK)
```